posydon/popsyn/synthetic_population.py

```
@@ -2038,10 +2035,6 @@ def calculate_cosmic_weights(self, SFH_identifier, MODEL_in=None):
         if MODEL_in is None:
             MODEL = DEFAULT_MODEL
         else:
-            for key in MODEL_in:
-                if key not in DEFAULT_MODEL:
-                    raise ValueError(key + " is not a valid parameter name!")
```

Instead of removing I'd throw a Warning here, that keys get ignored/might be unsupported.